### PR TITLE
HotFix: CustomUser is not JSON serializable

### LINKED ISSUE #29: [BUG]: CustomUser is not JSON serializable
## Describe the bug
Calling the GET All API returning the 500 telling that `CustomUser is not JSON serializable`

## Step To Reproduce the bug:

(1-3 is done in Admin page)
1. Create the CustomUser object.
2. Create the assignment then assign the author to the assignment with the created CustomUser
3. Call the GET All assignment API.
4. API Return 500 and the server crashed.

## Expected behavior

This should return the assignment object containing the `uid` instead of the entire CustomUser object.

## Error example
```
Internal Server Error: /api/v1/assignment/
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/django/core/handlers/base.py", line 204, in _get_response
    response = response.render()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/rest_framework/response.py", line 70, in rendered_content
    ret = renderer.render(self.data, accepted_media_type, context)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/rest_framework/renderers.py", line 100, in render
    ret = json.dumps(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/rest_framework/utils/json.py", line 25, in dumps
    return json.dumps(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 234, in dumps
    return cls(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/rest_framework/utils/encoders.py", line 67, in default
    return super().default(obj)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type CustomUser is not JSON serializable
[02/Nov/2021 16:37:06] "GET /api/v1/assignment/ HTTP/1.1" 500 129485
```

## Example data
Screen Shot 2564-11-02 at 23 40 28

### PR DESCRIPTION
**Pull request by abounk**
#29

## PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others. (Please describe)

### Fixes

1. Calling the GET All API returning the 500 telling that CustomUser is not JSON serializable. #29
2. tagIdField returns author's uid instead of author object.

### CODE REVIEW

**Comment by abounk** [comment]
🔧🔧🔧

File: garnbarn_api/serializer.py [hunk 1]
```diff
@@ -109,6 +118,7 @@ class CreateAssignmentApiSerializer(serializers.ModelSerializer):
     reminderTime = ReminderTimeField(source='reminder_time', default=None,
                                      child=serializers.IntegerField()
                                      )
+    tag = CreateTagApiSerializer(default=None)
```
Comment (sirateek) [inline garnbarn_api/serializer.py:121]: Users shouldn't specify the Tag object from the API. But instead, they have to specify Tag ID.

Currently, I can still specify the tag param in the create assignment API request. But the server return 500

Example: 
```
{
    "name": "Test1234",
    "description": "Here!",
    "tag": {
        "test": "A",
        "name": "1234"
    }
}
```

**Review by sirateek — CHANGES_REQUESTED** [review]
This change will also affect Create Assignment API. Since it is a hotfix, I can't accept it. Please only fix the bug.

File: garnbarn_api/serializer.py (hunk 1 above)
Comment (abounk) [reply garnbarn_api/serializer.py]: resolved with https://github.com/GarnBarn/garnbarn-backend/pull/30/commits/8e996c3dec4f89ae8582ce077278c6b538b07b70

**Review by sirateek — APPROVED** [review]
LGTM.

File: garnbarn_api/serializer.py (hunk 1 above)
Comment (abounk) [reply garnbarn_api/serializer.py]: Fixed.

**Pull request merged**